src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CandidateFileEntry.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.SourceBuild.Tasks.LeakDetection
+{
+    internal class CandidateFileEntry
```

Comment:
Given the usage of this class, consider making this a nested class of CheckForPoison.  Reasons for this are it is not intended to be used outside of that type.  It make this more clear and is generally nicer when refactoring IMO.

Reply:
Addressed here: https://github.com/dotnet/installer/pull/17438/commits/b99b930bf2726d24488910226e0fac3f3575e515